src/rlang/vec.h

```diff
@@ -339,7 +339,7 @@ bool r_is_string(r_obj* x) {
 static inline
 bool r_arg_as_bool(r_obj* x, const char* arg) {
   if (!r_is_bool(x)) {
-    r_abort("`%s` must be a logical value.", arg);
+    r_abort("`%s` must be `TRUE` or `FALSE`.", arg);
```

Comment:
By the way the rlang lib exports `r_format_lazy_error_arg()`, this returns a vmax-protected C string that is equivalent to `{.arg }` formatting.